onnxruntime/core/graph/graph.cc

```diff
@@ -1818,16 +1818,36 @@ void Graph::ReverseDFSFrom(gsl::span<const Node* const> from,
   }
 }
 
+template <typename T>
+struct VisitorPriorityQueue {
+  using ComparatorType = std::function<bool(T, T)>;
+  std::list<T> list_;
+  const ComparatorType comparator_ = nullptr;
+  VisitorPriorityQueue(const ComparatorType& comp) : comparator_(comp) {}
+
+  void push(T node) {
+    list_.insert(
+        std::upper_bound(list_.begin(), list_.end(), node, comparator_),
```

Comment:
Binary search on a linked list is of linear time complexity, but based on the data it looks like there were other factors contributing to the slowness of this queue namely insertions. Could've just used find() and got the same performance.

Reply:
@smk2007 Are you able to measure what's faster/simpler? This or find() or the approach in the other PR